### PR TITLE
Wav2 vec2 phoneme ctc tokenizer optimisation

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?

This is my FIRST PR! 
The Wav2Vec2PhonemCTCTokenizer is slow when its argument `do_phonemize` is set to True. It re-initialises the backend at each forward pass. This is adressed using a class argument. 

There was also an H4 title in the documentation which had a link which did not render( `<h4></h4>` used to replace `####`)  

Tests were passed, no additional ones were created. Runtime experiments to phonemize the entire 'tr' (turkish) subset of the common voice dataset gives a x10 boost in performances. 

Models:
- Wav2Vec2PhonemeCTCTokenizer: @patrickvonplaten, @LysandreJik 

Documentation: @sgugger

### CODE REVIEW

File: src/transformers/models/wav2vec2_phoneme/tokenization_wav2vec2_phoneme.py [hunk 1]
```diff
@@ -169,6 +172,17 @@ def vocab_size(self) -> int:
     def get_vocab(self) -> Dict:
         return dict(self.encoder, **self.added_tokens_encoder)
 
+    def init_backend(self, phonemizer_lang: str):
+        """Initiaizes the backend
```
Comment (sgugger) [inline src/transformers/models/wav2vec2_phoneme/tokenization_wav2vec2_phoneme.py:176]:
```suggestion
        """
        Initializes the backend.
```

File: src/transformers/models/wav2vec2_phoneme/tokenization_wav2vec2_phoneme.py [hunk 2]
```diff
@@ -169,6 +172,17 @@ def vocab_size(self) -> int:
...
+
+        Args:
+            phonemizer_lang (str): language to be used
```
Comment (sgugger) [inline src/transformers/models/wav2vec2_phoneme/tokenization_wav2vec2_phoneme.py:179]:
```suggestion
            phonemizer_lang (`str`): The language to be used.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -368,8 +368,7 @@ For documentation strings, 🤗 Transformers follows the [google style](https://
 Check our [documentation writing guide](https://github.com/huggingface/transformers/tree/main/docs#writing-documentation---specification)
 for more information.
 
-#### This guide was heavily inspired by the awesome [scikit-learn guide to contributing](https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md)
-
+<h4> This guide was heavily inspired by the awesome <a href=https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md>scikit-learn guide to contributing</a></h4>
```
Comment (sgugger) [inline CONTRIBUTING.md:371]: This doesn't need to be a header (there is no reason to make it one). Let's just put this in bold and keep Markdown syntax.

**Review by sgugger — APPROVED** [review]
Congrats on your first PR!
LGTM with a few nits, but let's wait for @patrickvonplaten approval before merging.

**Review by patrickvonplaten — APPROVED** [review]
Great! Thanks for fixing this :-)

File: CONTRIBUTING.md (hunk 3 above)
Comment (patrickvonplaten) [reply CONTRIBUTING.md:371]: +1

**Pull request merged**